support-lib/cpp/DataRef.cpp

```diff
@@ -324,6 +329,89 @@ DataRef::DataRef(CFDataRef platformObj) {
     _impl = std::make_shared<DataRefObjc>(platformObj);
 }
 
+#elif DATAREF_WASM
+
+class DataRefWasm : public DataRef::Impl {
```

Comment:
can we just put this into a separate file and include it when needed?   very large #if blocks seems problematic to maintain over time.